packages/core/test/endowmentsToolkit.spec.js

```diff
@@ -2,11 +2,12 @@ const test = require('ava')
 const endowmentsToolkit = require('../src/endowmentsToolkit.js')
 
 function prepareTest({ knownWritable } = {}) {
```

Comment:
```suggestion
/**
 * @param {Object} [options]
 * @param {Set<string>} [options].knownWritable]
 * @returns {ReturnType<typeof endowmentsToolkit>}
 */
function prepareTest({ knownWritable } = {}) {
```

- [complaining] I note that the return type of `endowmentsToolkit` probably needs to be explicitly defined somewhere because rn it's only inferred.

Reply:
I was told by an expert at a conference that the most common mistake people make is defining return types explicitly too often, while of you do typescript right it should infer all of them correctly.

I agree this type needs a name tho 😁

Reply:
with the caveat that explicit type definitions are necessary for [isolated declarations](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#isolated-declarations), though we don't need that atm